pyvrp/cpp/Route.cpp

```diff
@@ -81,6 +95,26 @@ Route::Route(ProblemData const &data, Visits visits, size_t const vehicleType)
     slack_ = ds.twLate() - ds.twEarly();
     timeWarp_ = ds.timeWarp(vehType.maxDuration);
     release_ = ds.releaseTime();
+
+    schedule_.reserve(size());
+    auto now = startTime_;
+    for (size_t prevClient = startDepot_; auto const client : visits_)
+    {
+        now += durations(prevClient, client);
+
+        ProblemData::Client const &clientData = data.location(client);
+        auto const wait = std::max<Duration>(clientData.twEarly - now, 0);
+        auto const timeWarp = std::max<Duration>(now - clientData.twLate, 0);
+
+        now += wait;
+        now -= timeWarp;
+
+        schedule_.emplace_back(
+            now, now + clientData.serviceDuration, wait, timeWarp);
+
+        now += clientData.serviceDuration;
+        prevClient = client;
+    }
```

Comment:
Ideally this'd be moved inside the main loop above.

Reply:
Yeah. I think we can merge this into one loop but it's a bit of work figuring out how that I don't know if it's worth spending time on right now.